### PR TITLE
user/zellij: new package

### PR DESCRIPTION
**Pull request by denisstrizhkin**
Zellij is a modern terminal multiplexer written in rust. Would be nice to have in cports.

https://zellij.dev/about/

### CODE REVIEW

File: contrib/zellij/template.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+pkgdesc = "Terminal workspace with batteries included"
+maintainer = "Denis Strizhkin <strdenis02@gmail.com>"
+license = "MIT"
+url = "https://github.com/zellij-org/zellij"
+source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "1f0bfa13f2dbe657d76341a196f98a3b4caa47ac63abee06b39883a11ca220a8"
+
+def post_build(self):
+    with open(self.cwd / "docs/MANPAGE.md", "rb") as i:
+        with open(self.cwd / f"docs/{pkgname}.1", "w") as o:
```
Comment (nekopsykose) [inline contrib/zellij/template.py:16]:
```suggestion
        with open(self.cwd / f"docs/zellij.1", "w") as o:
```

here and everywhere else, generally always want to expand pkgname

File: contrib/zellij/template.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    
+    
+def post_install(self):
+    self.install_man(self.cwd / f"docs/{pkgname}.1")
+    self.install_license(self.cwd / "LICENSE.md")
+
+    for shell in ["bash", "fish", "zsh"]:
+        self.install_completion(f"assets/{pkgname}.{shell}", shell, pkgname)
+
+    self.install_file("assets/logo.png", "usr/share/pixmaps", name=f"{pkgname}.png")
```
Comment (nekopsykose) [inline contrib/zellij/template.py:34]: pixmaps is a legacy location; it should go in /usr/share/icons/hicolor/$size/apps/ instead

File: contrib/zellij/template.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+pkgname = "zellij"
+pkgver = "0.40.1"
+pkgrel = 0
+build_style = "cargo"
+make_check_args = ["--release"]
+hostmakedepends = ["cargo-auditable", "pkgconf", "openssl-devel", "go-md2man"]
```
Comment (nekopsykose) [inline contrib/zellij/template.py:6]:
```suggestion
hostmakedepends = ["cargo-auditable", "pkgconf", "go-md2man"]
makedepends = ["rust-std", "openssl-devel"]
```

File: contrib/zellij/template.py (hunk 2 above)
Comment (nekopsykose) [reply contrib/zellij/template.py:34]: though for a 2269 x 2620 icon i'm not sure where exactly, maybe someone else knows :)

File: contrib/zellij/template.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                "setup", "--generate-completion", shell, stdout=o
+            )
+    
+    
+def post_install(self):
+    self.install_man(self.cwd / f"docs/{pkgname}.1")
+    self.install_license(self.cwd / "LICENSE.md")
+
+    for shell in ["bash", "fish", "zsh"]:
+        self.install_completion(f"assets/{pkgname}.{shell}", shell, pkgname)
```
Comment (triallax) [inline contrib/zellij/template.py:32]: `pkgname` is the default
```suggestion
        self.install_completion(f"assets/{pkgname}.{shell}", shell)
```

File: contrib/zellij/template.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        with open(self.cwd / f"assets/{pkgname}.{shell}", "w") as o:
+            self.do(
+                f"target/{self.profile().triplet}/release/{pkgname}",
+                "setup", "--generate-completion", shell, stdout=o
+            )
+    
+    
+def post_install(self):
+    self.install_man(self.cwd / f"docs/{pkgname}.1")
+    self.install_license(self.cwd / "LICENSE.md")
```
Comment (triallax) [inline contrib/zellij/template.py:29]:
```suggestion
    self.install_man(f"docs/{pkgname}.1")
    self.install_license("LICENSE.md")
```

File: contrib/zellij/template.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+pkgrel = 0
+build_style = "cargo"
+make_check_args = ["--release"]
+hostmakedepends = ["cargo-auditable", "pkgconf", "openssl-devel", "go-md2man"]
+pkgdesc = "Terminal workspace with batteries included"
+maintainer = "Denis Strizhkin <strdenis02@gmail.com>"
+license = "MIT"
+url = "https://github.com/zellij-org/zellij"
+source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "1f0bfa13f2dbe657d76341a196f98a3b4caa47ac63abee06b39883a11ca220a8"
```
Comment (triallax) [inline contrib/zellij/template.py:12]:
```suggestion
sha256 = "1f0bfa13f2dbe657d76341a196f98a3b4caa47ac63abee06b39883a11ca220a8"
# generates completions with host bin
options = ["!cross"]
```

File: contrib/zellij/template.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+pkgver = "0.40.1"
+pkgrel = 0
+build_style = "cargo"
+make_check_args = ["--release"]
+hostmakedepends = ["cargo-auditable", "pkgconf", "openssl-devel", "go-md2man"]
+pkgdesc = "Terminal workspace with batteries included"
+maintainer = "Denis Strizhkin <strdenis02@gmail.com>"
+license = "MIT"
+url = "https://github.com/zellij-org/zellij"
+source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
```
Comment (triallax) [inline contrib/zellij/template.py:11]:
```suggestion
url = "https://zellij.dev"
source = f"https://github.com/zellij-org/zellij/archive/refs/tags/v{pkgver}.tar.gz"
```

File: contrib/zellij/template.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+pkgname = "zellij"
+pkgver = "0.40.1"
+pkgrel = 0
+build_style = "cargo"
+make_check_args = ["--release"]
```
Comment (triallax) [inline contrib/zellij/template.py:5]: i think we don't want this?

**Review by triallax — COMMENTED** [review]
please format with ruff/black

File: contrib/zellij/template.py (hunk 8 above)
Comment (nekopsykose) [reply contrib/zellij/template.py:5]: it fails without it because of the way debug/release interacts with whether plugins are wasm or not

Comment (nekopsykose) [reply contrib/zellij/template.py:5]: feel free to remove and figure out the other way thought :P  

might also just make sense to skip tests for this, dunno if they test something useful

Comment (triallax) [reply contrib/zellij/template.py:5]: ah ok

Comment (triallax) [reply contrib/zellij/template.py:5]: would be nice to add a comment for that

**Comment by denisstrizhkin** [comment]
Thanks for pointing out issues with my template will remember them in the future ones.

File: contrib/zellij/template.py (hunk 2 above)
Comment (denisstrizhkin) [reply contrib/zellij/template.py]: I don't really know what to do with this icon...

**Comment by triallax** [comment]
it seems like wasmer (which stable zellij depends on) and wasmtime (which master zellij switched to from wasmtime) don't support all of our archs (namely, `ppc*`), so you will need to move this template to the `user` repository and set `archs = ["aarch64", "riscv64", "x86_64"]`

File: contrib/zellij/template.py (hunk 2 above)
Comment (nekopsykose) [reply contrib/zellij/template.py]: sure, let's keep it here for now

**Comment by nekopsykose** [comment]
also forgot to tell you to squash the commits, but you can remember that next time :)

**Pull request merged**